BUILD.md

```diff
@@ -1,39 +1,94 @@
+
+```bash
+mkdir ~/ydbwork && cd ~/ydbwork
+```
 
-## Prerequisites
+## Clone the ydb repository
 
-Below is a list of packages that need to be installed before building YDB. [How to Build](#how-to-build) section contains step by step instructions to obtain these packages.
+```bash
+git clone https://github.com/ydb-platform/ydb.git
+```
+
+Change directory to the cloned repository to perform further commands:
+
+```
+cd ydb
+```
+
+### Check out trunk
+
+By default, the `main` branch is checked out. It contains the trunk - a latest development update for both YDB Server and CLI, for __Yatool build__. To build trunk using __CMake build__, check out the `cmakebuild` branch:
+
+```bash
+git checkout cmakebuild
+```
+
+The `cmakebuild` branch is updated on hourly basis following the `main`.
+
+Trunk may sometimes be broken for short periods of time, so you may prefer to build the latest stable versions of YDB Server and CLI as described below. As stable versions of the YDB Server and CLI belong to different commits, it is not possible to get both server and CLI stable executables with a single checkout/build. Checkout and build server first, then CLI, or visa versa.
+
+Release tags support both Yatool and CMake builds.
+
+### Check out the latest stable YDB Server version for build
+
+To build the latest stable version of a YDB Server, check out the latest stable Server tag from the repository. To do so, visit the https://github.com/ydb-platform/ydb/releases/latest page and use the provided 'tag' value in the `git checkout <tag>` command.
 
- - cmake 3.22+
- - clang-14
- - lld-14
- - git 2.20+
- - python3.8
- - pip3
- - antlr3
- - libaio-dev
- - libidn11-dev
- - ninja 1.10+
+For example, at the time of writing the latest stable release was [YDB Server 23.2.12](https://github.com/ydb-platform/ydb/releases/tag/23.2.12) with a tag `23.2.12`, so the command looks like that:
+
+```bash
+git checkout 23.2.12
+```
+
+### Check out the latest stable YDB CLI version for build
+
+To build a latest stable version of a YDB CLI, check out the latest stable CLI tag from the repository. To do so, visit the https://github.com/ydb-platform/ydb/releases page, scroll down to the top-most 'YDB CLI' release, and use the provided 'tag' value in the `git checkout <tag>` command.
+
+For example, at the time of writing the latest YDB CLI release was [YDB CLI 2.5.0](https://github.com/ydb-platform/ydb/releases/tag/CLI_2.5.0) with a tag `CLI_2.5.0`, so the command looks like that:
+
+```bash
+git checkout CLI_2.5.0
+```
+
+## Build using Yatool
+
+You can use native Yatool build for x86_64 under Linux Ubuntu 18+.
+
+Run `ya make` from the repository root, followed by an optional relative path to the build target. By default, `ya make` compiles a `debug` configuration, while you can choose to have `release` or `relwithdebinfo`. The latter is recommended for faster builds, as is leverages the YDB remote build cache.
```

Comment:
it'd be useful to clarify what's the difference between the mentioned build modes and when to use which

Reply:
Can we leave it for later fixes?